### DISCUSSION TITLE
Decorated async function not recognized as async task. Should this be fixed?

### DISCUSSION THREAD

**Discussion by ThomasAitken** [fastapi/fastapi#11089] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# utils/core.py

async def aiohttp_session_decorator(func):
    @wraps(func)  # Use wraps to preserve the signature and metadata of func
    async def wrapper(*args, **kwargs):
        async with aiohttp.ClientSession() as session:
            # Pass aiohttp session to the function if it's required
            return await func(session, *args, **kwargs)

    return wrapper

# tasks.py
@aiohttp_session_decorator
async def my_task(aiohttp_session: ClientSession, random_arg):
    ...

# router.py
    background_tasks.add_task(my_task, random_arg)
```

### Description

The implementation of `is_async_callable` that determines whether the task is run sync or async fails in this case because the decorator function doesn't return a co-routine. It tries to run this task synchronously which of course ultimately fails with `TypeError: 'coroutine' object is not callable`. There doesn't seem to be any obvious workaround.

Should we make the `is_async_callable` function handle this case of decorated async functions or should I just drop the use of top-level decorators for my async tasks?

### Operating System

Linux

### FastAPI Version

0.103.2

### Pydantic Version

2.6.0

### Python Version

3.11.4

**Comment by YuriiMotov** [comment]
Just fix the decorator (it should be just `def`, not `async def`) and everything works well:

```python
def aiohttp_session_decorator(func):
    @wraps(func)  # Use wraps to preserve the signature and metadata of func
    async def wrapper(*args, **kwargs):
        async with aiohttp.ClientSession() as session:
            # Pass aiohttp session to the function if it's required
            return await func(session, *args, **kwargs)

    return wrapper
```
